app_feup/lib/view/Widgets/print_info_card.dart

```diff
@@ -47,6 +77,55 @@ class PrintInfoCard extends GenericCard {
     );
   }
 
+  /// Returns a widget with all the print balance movements.
+  Widget printBalanceMovementsWidget(movements, context) {
+    if (movements.length <= 0) {
+      return empty('Não existem movimentos recentes', context);
+    }
+    return Column(
+      mainAxisSize: MainAxisSize.min,
+      children: this.printBalanceMovements(context, movements),
+    );
+  }
+
+  /// Returns a list of widgets with movements to be displayed in the print card
+  List<Widget> printBalanceMovements(context, movements) {
+    final List<Widget> rows = movements.map<Widget>((movement) {
+      return this.balanceMovement(context, movement);
+    }).toList();
+
+    return rows.length <= 3 ? rows : rows.sublist(0, 3);
+  }
+
+  // Individual movement row
+  Widget balanceMovement(context, movement) {
+    return Container(
+      margin: EdgeInsets.only(top: 8),
+      child: RowContainer(
+        color: Theme.of(context).backgroundColor,
+        child: Container(
+          padding: EdgeInsets.all(11),
+          child: Row(
+              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+              mainAxisSize: MainAxisSize.max,
+              crossAxisAlignment: CrossAxisAlignment.center,
+              children: <Widget>[
+                Text(DateTime.parse(movement['datetime']).formatter()),
+                Text(
+                  movement['value'][0] != '-' //it is positive, add + character
+                      ? '+' + movement['value']
+                      : movement['value'],
+                  style: Theme.of(context).textTheme.headline2.apply(
+                      color: movement['value'][0] == '-' //check if its negative
+                          ? Theme.of(context).textTheme.headline3.color
+                          : Colors.green),
+                ),
```

Comment:
Can you check for the positivity before returning the container, and just check for the boolean value here?

Reply:
Like this?
```suggestion
                movement['value'][0] != '-'
                  ? Text('+' + movement['value'],
                      style: Theme.of(context).textTheme.headline2.apply(
                        color: Colors.green)
                    )
                  : Text(movement['value'],
                      style: Theme.of(context).textTheme.headline2.apply(
                        color: Theme.of(context).textTheme.headline3.color)
                    )
```

Reply:
He meant to save the movement['value'][0] != '-' to a boolean variable, and then check it in the ternary operator, instead of repeating the operation, to make it more readable, and avoid repetition

Reply:
Although I think I like the suggested version more

Reply:
Yes, I was suggesting what @brunogomes30 was explaining, but I think that your solution is fine too.